compiler-rt/lib/ctx_profile/tests/CtxInstrProfilingTest.cpp

```diff
@@ -179,13 +179,15 @@ TEST_F(ContextTest, Dump) {
   (void)Subctx;
   __llvm_ctx_profile_release_context(&Root);
 
-  struct Writer {
+  struct Writer : public ProfileWriter {
```

Comment:
Its a bit strange to have Writer inherit from ProfileWriter. To me it makes more sense to call the interface `Writer` and the derived class / struct profile writer here. Wdyt?

Also a struct derived from a class seems strange to me.

Reply:
`Writer` is too generic. I only use it here because this is a test. I can rename it to `TestProfileWriter` (and make it a class, that *is* weird, indeed)